### PR TITLE
[Medallia] Add functionality for subpaths in Medallia

### PR DESCRIPTION
**Pull request by aubreyarcangel**
## Summary

This ticket updates the liquid filter 'getSurvey' to recognize subpaths of a URL if a custom Medallia survey exists for that URL.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#75521

## Testing done

Medallia cannot launched via local build due to permissions, so it must be tested on the review instance.

-The `getSurvey` liquid filter previously required subpaths to be explicitly defined in the Medallia survey JSON object.
- All unit tests were updated 

Please follow the steps below:

Test returning custom survey with subpaths

1. Go to the page `/health-care/eligibility`.
2. Click on the FEEDBACK button.
3. The modal for survey `41` should appear.

Test returning custom survey number without subpaths

1. Go to the page `/search`.
2. Click on the FEEDBACK button.
3. The modal for survey `20` should appear.

Test returning default survey:

1. Go to the homepage `/`.
2. Click on the FEEDBACK button.
3. The modal for survey `11` should appear.

## What areas of the site does it impact?
Any page with a Medallia FEEDBACK button at the bottom of the main content

## Acceptance criteria

- [x] Given a valid URL that matches an entry in either surveyData.urls or surveyData.urlsWithSubPaths, then it should return the correct survey ID associated with the URL
- [x] Given a URL that does not directly match an entry in surveyData.urls but is a subpath of an entry in surveyData.urlsWithSubPaths, then it should return the correct survey ID associated with the URL
- [x] Given a URL that does not match any entry in surveyData.urls or any subpath in surveyData.urlsWithSubPaths, then it should return a default survey ID

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

**Review by micahchiang — COMMENTED** [review]
Have you considered storing these numbers as constants, instead of hardcoding them across multiple files? My concern is that they'll need to be updated eventually and instances will be missed.

**Comment by aubreyarcangel** [comment]
> Have you considered storing these numbers as constants, instead of hardcoding them across multiple files? My concern is that they'll need to be updated eventually and instances will be missed.

@micahchiang Thanks for the feedback. I refactored and stored the survey numbers as constants.

File: src/site/assets/js/medallia/vagovstaging.js [hunk 1]
```diff
@@ -8,21 +7,86 @@
...
+      staging: SURVEY_NUMBERS.HEALTH_CARE_STAGING,
+    },
+    '/my-health/medical-records/summaries-and-notes/visit-summary': {
+      production: SURVEY_NUMBERS.VISIT_SUMMARY_PROD,
+      staging: SURVEY_NUMBERS.VISIT_SUMMARY_STAGING,
+    },
+  },
+};
+
+function getSurvey(buildtype, url) {
```
Comment (randimays) [inline src/site/assets/js/medallia/vagovstaging.js:59]: This seems to be a duplicate of the liquid filter. Should this be here?

Comment (aubreyarcangel) [reply src/site/assets/js/medallia/vagovstaging.js:59]: That was intentional. Initially I tried importing the getSurey function and survey data via `const { SURVEY_NUMBERS, medalliaSurveys } = require('../../../filters/medalliaSurveysConfig')`, but learned that didn't work and would have to make some changes to the webpack config to share that code, so instead I opted for the quick and easy route for right now. Currently I'm only using it for debugging and adding it on `localhost` and `review instances` only via [src/site/includes/survey-tools.html](https://github.com/department-of-veterans-affairs/content-build/blob/cd9fd9f60da654c50418243139d5c0a4bb7e7180/src/site/includes/survey-tools.html#L33)

Comment (randimays) [reply src/site/assets/js/medallia/vagovstaging.js:59]: Gotcha 👍🏻 Thanks for that clarification

**Review by randimays — APPROVED** [review]

**Comment by Andrew565** [comment]
PR has a failing required test, opting not to review at this time. - Tier 2/FE CoP

**Review by Andrew565 — APPROVED** [review]

**Pull request merged**